tests/TaskSerializer/DataviewTaskSerializer.test.ts

```diff
@@ -0,0 +1,276 @@
+import { DATAVIEW_SYMBOLS, DataviewTaskSerializer } from '../../src/TaskSerializer/DataviewTaskSerializer';
+import type { Task } from '../../src/Task';
+import type { TaskDetails } from '../../src/TaskSerializer';
+
+jest.mock('obsidian');
+window.moment = moment;
+
+describe('DataviewTaskSerializer', () => {
+    const taskSerializer = new DataviewTaskSerializer();
+    const serialize = taskSerializer.serialize.bind(taskSerializer);
+    const deserialize = taskSerializer.deserialize.bind(taskSerializer);
+
+    const dateFields = ['startDate', 'dueDate', 'doneDate', 'createdDate', 'scheduledDate'] as const;
+
+    describe('deserialize', () => {
+        it('should parse an empty string', () => {
+            const taskDetails = deserialize('');
+            expect(taskDetails).toMatchTaskDetails({});
+        });
+
+        it.each(dateFields)('should parse a %s', (field) => {
+            const symbol = DATAVIEW_SYMBOLS[`${field}Symbol`];
+            const taskDetails = deserialize(`[${symbol} 2021-06-20]`);
+            expect(taskDetails).toMatchTaskDetails({ [field]: moment('2021-06-20', 'YYYY-MM-DD') });
+        });
+
+        it('should parse a priority', () => {
+            const priorities = ['High', 'Medium', 'Low'] as const;
+            for (const p of priorities) {
+                const prioritySymbol = DATAVIEW_SYMBOLS.prioritySymbols[p];
+                const priority = Priority[p];
+
+                const taskDetails = deserialize(`[${prioritySymbol}]`);
+
+                expect(taskDetails).toMatchTaskDetails({ priority });
+            }
+        });
+
+        it('should parse a recurrence', () => {
+            const taskDetails = deserialize('[repeat:: every day]');
+            expect(taskDetails).toMatchTaskDetails({
+                recurrence: new RecurrenceBuilder().rule('every day').build(),
+            });
+        });
+
+        it('should parse a task with multiple fields and tags', () => {
+            const taskDetails = deserialize(
+                'Wobble [priority::high] #tag1 [completion::2022-07-02] #tag2  [due::2022-07-02] #tag3 [scheduled::2022-07-02] #tag4 [start::2022-07-02] #tag5  [repeat::every day]  #tag6 #tag7 #tag8 #tag9 #tag10',
+            );
+
+            expect(taskDetails).toMatchTaskDetails({
+                description: 'Wobble #tag1 #tag2 #tag3 #tag4 #tag5 #tag6 #tag7 #tag8 #tag9 #tag10',
+                dueDate: moment('2022-07-02', 'YYYY-MM-DD'),
+                doneDate: moment('2022-07-02', 'YYYY-MM-DD'),
+                startDate: moment('2022-07-02', 'YYYY-MM-DD'),
+                scheduledDate: moment('2022-07-02', 'YYYY-MM-DD'),
+                priority: Priority.High,
+                recurrence: new RecurrenceBuilder()
+                    .rule('every day')
+                    .dueDate('2022-07-02')
+                    .scheduledDate('2022-07-02')
+                    .startDate('2022-07-02')
+                    .build(),
```

Comment:
I was all set to merge this, then noticed that this new test has a lot of 'symmetry' in it.

That is, by using the same date for all date fields, it isn't testing whether the date values are read in to the correct fields, which makes it a bit weak.

Could each date type have a different value please?